src/Dashboard/Dashboard.js

```diff
@@ -23,18 +22,19 @@ class Dashboard extends Component {
         this.setDashboard();
     }
 
+    getDashboard = () => {
+        switch (this.props.mode) {
+            case 'edit':
+                return <EditDashboard />;
+            case 'new':
+                return <NewDashboard />;
+            default:
+                return <ViewDashboard />;
+        }
+    };
+
     render() {
-        return this.props.mode === 'view' ? (
-            <Fragment>
-                <DashboardsBar />
-                <DashboardVerticalOffset />
-                <ViewDashboardContent />
-            </Fragment>
-        ) : (
-            <Fragment>
-                <EditDashboard mode={this.props.mode} />
-            </Fragment>
-        );
+        return <Fragment>{this.getDashboard()}</Fragment>;
```

Comment:
This Fragment is not necessary

Reply:
And now that render() is purely a function call it is probably over-extracted :-)